pkg/machine/compression/gzip.go

```diff
@@ -0,0 +1,36 @@
+package compression
+
+import (
+	"compress/gzip"
+	"io"
+
+	crcOs "github.com/crc-org/crc/v2/pkg/os"
+	"github.com/sirupsen/logrus"
+)
+
+type gzipDecompressor struct {
+	genericDecompressor
+	gzReader io.ReadCloser
+}
+
+func newGzipDecompressor(compressedFilePath string) (*gzipDecompressor, error) {
+	d, err := newGenericDecompressor(compressedFilePath)
+	return &gzipDecompressor{*d, nil}, err
+}
+
+func (d *gzipDecompressor) decompress(w io.WriteSeeker, r io.Reader) error {
+	gzReader, err := gzip.NewReader(r)
+	if err != nil {
+		return err
+	}
+	d.gzReader = gzReader
+	_, err = crcOs.CopySparse(w, gzReader)
```

Comment:
(Non-blocking: I’m tempted to say that code like this doesn’t need to be a subclass; parametrize `genericDecompressor` with a
```go
    decompressStream func(io.Reader) (io.ReadCloser, error)
    isSparse bool
```

… but then I would go on and argue that “is the output a sparse file” should be true either for all or no compression formats because it is very surprising for the sparseness of the output to depend on the compression algorithm … and we would drift into things that should happen in other PRs.)